File: LATIHAN2/Latihan2_Python/Memory.py
from Hardware import Hardware


class Memory(Hardware):
    # class attribute
    __frequency = 0
    __memorySize = 0
    __supportCuda = ""

    # constructors
    def __init__(self, frequency, memorySize, supportCuda):
        self.__frequency = frequency
        self.__memorySize = memorySize
        self.__supportCuda = supportCuda

    # getter and setter
    def setFrequency(self, frequency):
        self.__frequency = frequency

    def getFrequency(self):
        return self.__frequency

    def setMemorySize(self, memorySize):
        self.__memorySize = memorySize

    def getMemorySize(self):
        return self.__memorySize

    def setSupportCuda(self, supportCuda):
        self.__supportCuda = supportCuda

    def getSupportCuda(self):
        return self.__supportCuda

    # method for showing inside class
    def ShowClass(self):
        print("Id Product	 : " + self.getIdProduct())
        print("Price		 : " + str(self.getPrice()))
        print("Brand		 : " + self.getBrand())
        print("Model		 : " + self.getModel())
        print("Frequency Memory : " + str(self.getFrequency()))
        print("Memory Size	 : " + str(self.getMemorySize()))
        print("Support Cuda	 : " + self.getSupportCuda())

File: LATIHAN2/Latihan2_Java/Memory.java
public class Memory extends Hardware {

	private String frequency;
	private String memorySize;
	private String supportsCuda;

	// empty constructors
	Memory() {

	}

	// constructors
	Memory(String frequency, String memorySize, String supportsCuda) {
		this.frequency = frequency;
		this.memorySize = memorySize;
		this.supportsCuda = supportsCuda;
	}

	// getter and setter
	public void setFrequency(String frequency) {
		this.frequency = frequency;
	}

	public String getFrequency() {
		return frequency;
	}

	public void setMemorySize(String memorySize) {
		this.memorySize = memorySize;
	}

	public String getMemorySize() {
		return memorySize;
	}

	public void setSupportsCuda(String supportsCuda) {
		this.supportsCuda = supportsCuda;
	}

	public String getSupportsCuda() {
		return supportsCuda;
	}

	// method for showing inside class
	public void ShowClass() {
		System.out.println("Id Product	 : " + this.getIdProduct());
		System.out.println("Price	  	 : " + this.getPrice());
		System.out.println("Brand	 	 : " + this.getBrand());
		System.out.println("Model	 	 : " + this.getModel());
		System.out.println("Frequency Memory : " + this.getFrequency());
		System.out.println("Memory Size	 : " + this.getMemorySize());
		System.out.println("Support Cuda	 : " + this.getSupportsCuda());
	}

}
